src/lib/viz/widgets/widget.ts

```diff
@@ -0,0 +1,44 @@
+import { DataView } from '@/data/dataview/dataview';
+import { CartoError } from '@/core/errors/CartoError';
+import { queryDOMElement } from '../utils/dom';
+
+export class Widget {
```

Comment:
Should this be an abstract class?

Reply:
Yes, gonna change it right now and merge